### ISSUE TITLE
Sidecar egress port results in listener with BlackHoleCluster filter

### ISSUE THREAD

**Issue by mdgreenfield** [istio/istio#12536]
**Describe the bug**
Configuring `Sidecar` `egress` ports for namespaces other than `istio-system` results in a `envoy.tcp_proxy` filter of `BlackHoleCluster`

**Expected behavior**
The Envoy outbound cluster configuration should be set to the correct cluster.

**Steps to reproduce the bug**
1. Configure a `Sidecar` with the configuration:

    ```yaml
    - hosts:
        - <namespace>/<service>.<namespace>.svc.mesh.local
        port:
          name: tcp-<service>
          number: 5757
          protocol: TCP
    ```
1. Inspect the Envoy's configuration (i.e. `localhost:15000/config_dump`) and notice that this particular listener has a `BlackHoleCluster` filter in its chain.

After doing the above, remove the `port` configuration and notice that the correct outbound listener is configured.

**Version**
```
$ ./istioctl version --remote
client version: version.BuildInfo{Version:"1.1.0-rc.5", GitRevision:"319ae418ed6564e4323b19d29a679f6f9234f829", User:"root", Host:"c0df9197-461a-11e9-b4f5-0a580a2c0404", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Clean", GitTag:"1.1.0-rc.4-7-g319ae41"}
citadel version: version.BuildInfo{Version:"1.1.0-rc.5", GitRevision:"319ae418ed6564e4323b19d29a679f6f9234f829-dirty", User:"root", Host:"c0df9197-461a-11e9-b4f5-0a580a2c0404", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.0-rc.4-7-g319ae41"}
galley version: version.BuildInfo{Version:"1.1.0-rc.5", GitRevision:"319ae418ed6564e4323b19d29a679f6f9234f829-dirty", User:"root", Host:"c0df9197-461a-11e9-b4f5-0a580a2c0404", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.0-rc.4-7-g319ae41"}
pilot version: version.BuildInfo{Version:"1.1.0-rc.5", GitRevision:"319ae418ed6564e4323b19d29a679f6f9234f829-dirty", User:"root", Host:"c0df9197-461a-11e9-b4f5-0a580a2c0404", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.0-rc.4-7-g319ae41"}
sidecar-injector version: version.BuildInfo{Version:"1.1.0-rc.5", GitRevision:"319ae418ed6564e4323b19d29a679f6f9234f829-dirty", User:"root", Host:"c0df9197-461a-11e9-b4f5-0a580a2c0404", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.0-rc.4-7-g319ae41"}
telemetry version: version.BuildInfo{Version:"1.1.0-rc.5", GitRevision:"319ae418ed6564e4323b19d29a679f6f9234f829-dirty", User:"root", Host:"c0df9197-461a-11e9-b4f5-0a580a2c0404", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.0-rc.4-7-g319ae41"}
```
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-04T04:48:03Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:25:46Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Installation**
We have a primary Istio cluster for only the control plane and two remote clusters for our workloads.

**Environment**
AWS running on Ubuntu VMs

/cc @howardjohn @costinm @hmakkapati

**Comment by howardjohn** [comment]
Cluster is set here https://github.com/istio/istio/blob/89f7d762ee635abd90b61321174383fbf37c8555/pilot/pkg/networking/core/v1alpha3/tls.go#L258-L266 because service is nil

Service is always nil because it is never set here https://github.com/istio/istio/blob/effeb0f9302054b13eccf9c1339fcd3f8b1f3ab0/pilot/pkg/networking/core/v1alpha3/listener.go#L768-L777

**Comment by rshriram** [comment]
does the service listen on same port as 5757?

**Comment by howardjohn** [comment]
I believe it is always blackhole cluster if the port is there. Take a look at the code I linked above. But when I was testing I set service as 5757 and something else and got same results

**Comment by howardjohn** [comment]
@mdgreenfield after talking with Shriram we think this is a bug, but the fix is non-trivial.

You mentioned this is not a blocker, right? I will have a change to take a closer look at this in a few days.

**Comment by mdgreenfield** [comment]
Hemanth and I chatted about this. I don't think this is an immediate blocker for us but it is definitely important to us over the next couple weeks. If you've got time to look at it more closely in a couple days that should be fine.

**Comment by howardjohn** [comment]
Fix is merged, will be in 1.1.3. If this doesnt fix the issue please reopen
